core/include/thread.h

```diff
@@ -526,6 +537,81 @@ static inline bool thread_is_active(const thread_t *thread)
  */
 const char *thread_state_to_string(thread_status_t state);
 
+/**
+ * Get start address (lowest) of a thread's stack.
+ *
+ * @param   thread thread to work on
+ * @returns current stack pointer, or NULL if not available
+ */
+static inline void *thread_get_stackstart(const thread_t *thread)
+{
+#if defined(DEVELHELP) || defined(SCHED_TEST_STACK) \
+    || defined(MODULE_MPU_STACK_GUARD)
+    return thread->stack_start;
+#else
+    (void)thread;
+    return NULL;
+#endif
+}
+
+/**
+ * Get stored Stack Pointer of thread.
+ *
+ * *Only provides meaningful value if the thread is not currently running!*.
+ *
+ * @param   thread thread to work on
+ * @returns current stack pointer
+ */
+static inline void *thread_get_sp(const thread_t *thread)
+{
+    return thread->sp;
+}
+
+/**
+ * Get size of a thread's stack.
+ *
+ * @param   thread thread to work on
+ * @returns thread stack size, or 0 if not available
+ */
+static inline size_t thread_get_stacksize(const thread_t *thread)
+{
+#if defined(DEVELHELP)
+    return thread->stack_size;
+#else
+    (void)thread;
+    return 0;
+#endif
+}
+
+/**
+ * Get PID of thread.
+ *
+ * This is a simple getter for thread->pid.
+ *
+ * @param   thread thread to work on
+ * @returns thread pid
+ */
+static inline kernel_pid_t thread_get_pid(const thread_t *thread)
+{
+    return thread->pid;
+}
```

Comment:
How about

```suggestion
static inline kernel_pid_t thread_get_pid(const thread_t *thread)
{
    extern volatile kernel_pid_t sched_active_pid;

    if (thread) {
        return thread->pid;
    } else {
        return sched_active_pid;
    }
}
```

Reply:
hm, not sure. the dual name is an issue, but I'd rather not change "thread_getpid()". (that would be the reason for overloading thread_get_pid(), right?)

Reply:
Maybe name it `thread_getpid_of()`? That name would make it more obvious which of the two requires an extra argument.